### PR TITLE
Exact idl to tel

### PR DESCRIPTION
**Pull request by Johannes-Sahlmann**
Implement improved handling of idl to tel transformations.

### CODE REVIEW

**Comment by Johannes-Sahlmann** [comment]
This is a quite extensive PR that introduces several capabilities related to the implementation of exact 3D rotation transformations. Documentation is probably insufficient, but there is a draft ISR that explains the application to HST focal plane alignment.

**Comment by Johannes-Sahlmann** [comment]
@ColinRCox can you please have a look at this PR and let me know what you think. thanks!

File: pysiaf/utils/rotations.py [hunk 1]
```diff
@@ -401,32 +583,229 @@ def unit(ra, dec):
...
-    v3 = 3600*np.degrees(np.arcsin(u[2]/norm))
+    assert len(vector) == 3, 'Not a vector'
+
+    norm = np.sqrt(vector[0]**2 + vector[1]**2 + vector[2]**2)
+    v2 = 3600. * np.degrees(np.arctan2(vector[1], vector[0]))  # atan2 puts it in the correct quadrant
+    v3 = 3600. * np.degrees(np.arcsin(vector[2]/norm))
     return v2, v3
+
+
+def polar_angles(vector, positive_azimuth=False):
```
Comment (shanosborne) [inline pysiaf/utils/rotations.py]: Add the positive_azimuth parameter to the docstring

File: pysiaf/utils/rotations.py [hunk 2]
```diff
@@ -401,32 +583,229 @@ def unit(ra, dec):
...
+    Returns
+    -------
+    nu2, nu3 : tuple of floats with astropy quantity
+        The same position represented by polar coordinates
+
+    """
+    if len(vector) != 3:
+        raise ValueError('Input is not a vector or an array of vectors')
+
+    # norm = np.linalg.norm(vector, axis=1)
```
Comment (shanosborne) [inline pysiaf/utils/rotations.py]: Remove old line

**Comment by Johannes-Sahlmann** [comment]
@ColinRCox and @shanosborne : Thank you for your reviews. I addressed them and will merge when the tests pass.

**Pull request merged**